_ml-commons-plugin/gpu-acceleration.md

```diff
@@ -0,0 +1,285 @@
+# Install OS headers
+sudo yum install kernel-devel-$(uname -r) kernel-headers-$(uname -r) -y
+# Install Neuron Driver
+####################################################################################
+# Warning: If Linux kernel is updated as a result of OS package update
+#          Neuron driver (aws-neuron-dkms) should be re-installed after reboot
+####################################################################################
+sudo yum install aws-neuronx-dkms -y
+# Install Neuron Tools
+sudo yum install aws-neuronx-tools -y
+
+# Install Python venv and activate Python virtual environment to install    
+# Neuron pip packages.
+cd ~
+sudo yum install -y python3.7-venv gcc-c++
+python3.7 -m venv pytorch_venv
+source pytorch_venv/bin/activate
+pip install -U pip
+
+# Set Pip repository  to point to the Neuron repository
+pip config set global.extra-index-url https://pip.repos.neuron.amazonaws.com
+
+#Install Neuron PyTorch
+pip install torch-neuron torchvision
+# If need to trace neuron model, install torch neuron with this command
+# pip install torch-neuron neuron-cc[tensorflow] "protobuf<4" torchvision
+
+# If need to trace neuron model, install transformers for tracing Huggingface model.
+# pip install transformers
+
+# Copy torch neuron lib to OpenSearch
+PYTORCH_NEURON_LIB_PATH=~/pytorch_venv/lib/python3.7/site-packages/torch_neuron/lib/
+mkdir -p $OS_HOME/lib/torch_neuron; cp -r $PYTORCH_NEURON_LIB_PATH/ $OS_HOME/lib/torch_neuron
+export PYTORCH_EXTRA_LIBRARY_PATH=$OS_HOME/lib/torch_neuron/lib/libtorchneuron.so
+echo "export PYTORCH_EXTRA_LIBRARY_PATH=$OS_HOME/lib/torch_neuron/lib/libtorchneuron.so" | tee -a ~/.bash_profile
+# Increase JVm stack size to >=2MB
+echo "-Xss2m" | tee -a $OS_HOME/config/jvm.options
+# Increase max file descriptors to 65535
+echo "$(whoami) - nofile 65535" | sudo tee -a /etc/security/limits.conf
+# max virtual memory areas vm.max_map_count to 262144
+sudo sysctl -w vm.max_map_count=262144
+```
+
+## (AWS Inferntia) Install GPU accelerator drivers manually
+
+If the previous two scripts do not provision your GPU accelerated node properly, you can install the drivers for AWS Inferentia manually.
+
+1. Deploy an AWS accelerator instance based on your chosen Linux operating system. For instructions, [Deploy on AWS accelerator instance](https://awsdocs-neuron.readthedocs-hosted.com/en/latest/frameworks/torch/torch-neuron/setup/pytorch-install.html#deploy-on-aws-ml-accelerator-instance).
+
+2. (Optional) To monitor GPU usage of your accelerator instance, install [Neuron tools](https://awsdocs-neuron.readthedocs-hosted.com/en/latest/frameworks/torch/torch-neuron/setup/pytorch-install.html), which allows models to be used inside your instance.
+
+  ```
+  # Install Neuron Tools
+  sudo apt-get install aws-neuronx-tools -y
+
+  export PATH=/opt/aws/neuron/bin:$PATH
+
+  # Test 
+  neuron-top
+  ```
+
+3. Copy the Neuron library into OpenSearch.
+
+  ```
+  OS_HOME=<OpenSearch installation path>
```

Comment:
Break these out into individual steps. You start with an env variable declaration but you don't tell the user what it's for. Also, when using <> for parameters there should be no whitespace between them. Make it something like `opensearchInstallationPath` or `opensearch-installation-path` and use all lowercase unless you're doing camel case.

Also, I'm pretty sure `OS_HOME` is wrong. If you do a search in the core repo it has 0 hits so it doesn't exist. The environment variable is `OPENSEARCH_HOME`.

I would take that back for technical review to double check..

Reply:
Switched to `OPENSEARCH_HOME`.